### PR TITLE
Adds delayed cover loading

### PR DESCRIPTION
**Pull request by FL550**
This modification changes the way the covers are loaded. First a placeholder image is displayed and when the page is done with loading, the placeholder images are relaced by the high resolution cover images.

What do you think of this? This would be nice for slow loading connections, as right now it blocks displaying until all covers are completly loaded.

### CODE REVIEW

**Comment by FL550** [comment]
Apparently spotify has changed the order of high-res images

File: src/spotify-card.ts [hunk 1]
```diff
@@ -76,9 +76,18 @@ export class SpotifyCard extends LitElement {
     this.lib.connectedCallback();
   }
 
-  protected updated(changedProps: PropertyValues): void {
+  protected async updated(changedProps: PropertyValues): Promise<void> {
     super.updated(changedProps);
     this.lib.updated(this.hass);
+    await this.updateComplete.then(() => {
```
[the remark this answers is not in the document]
Comment (fondberg) [reply src/spotify-card.ts:82]: Await and .then() do the same thing. So you can either move the code in .then() after the await line or remove await

[the remark this answers is not in the document]
Comment (FL550) [reply src/spotify-card.ts:82]: Ah, good to know!

File: src/spotify-card.ts [hunk 2]
```diff
@@ -225,13 +234,16 @@ export class SpotifyCard extends LitElement {
...
-              />
-            </svg>`}
+        <div
+          class="cover"
+          data-spotify-image-url="${item.images.length > 0 ? item.images[0].url : ''}"
```
Comment (FL550) [inline src/spotify-card.ts:239]: The data-attribute is set here. With this, the images are loaded when the main ui is done with loading. This way, the first page load is much smoother as the size of the card stays the same. Mainly in list-view this is a benefit as the available playlist are directly shown. In img src the images are directly loaded and not after main ui rendering.

**Review by fondberg — APPROVED** [review]
Cool!

**Pull request merged**